test/new-e2e/tests/cws/fargate/fargate_test.go

```diff
@@ -0,0 +1,471 @@
+						SecretOptions: ecsx.TaskDefinitionSecretArray{
+							ecsx.TaskDefinitionSecretArgs{
+								Name:      pulumi.String("apikey"),
+								ValueFrom: apiKeyParam.Name,
+							},
+						},
+					},
+				},
+				"log_router": {
+					Cpu:       pulumi.IntPtr(0),
+					User:      pulumi.StringPtr("0"),
+					Name:      pulumi.String("log_router"),
+					Image:     pulumi.String("amazon/aws-for-fluent-bit:latest"),
+					Essential: pulumi.BoolPtr(true),
+					FirelensConfiguration: ecsx.TaskDefinitionFirelensConfigurationArgs{
+						Type: pulumi.String("fluentbit"),
+						Options: pulumi.StringMap{
+							"enable-ecs-log-metadata": pulumi.String("true"),
+						},
+					},
+				},
+				"cws-instrumentation-init": {
+					Cpu:       pulumi.IntPtr(0),
+					Name:      pulumi.String("cws-instrumentation-init"),
+					Image:     pulumi.String(cwsCfg.getFullImagePath()),
+					Essential: pulumi.BoolPtr(false),
+					Command: pulumi.ToStringArray([]string{
+						"/cws-instrumentation",
+						"setup",
+						"--cws-volume-mount",
+						"/cws-instrumentation-volume",
+					}),
+					MountPoints: ecsx.TaskDefinitionMountPointArray{
+						ecsx.TaskDefinitionMountPointArgs{
+							SourceVolume:  pulumi.String("cws-instrumentation-volume"),
+							ContainerPath: pulumi.String("/cws-instrumentation-volume"),
+							ReadOnly:      pulumi.Bool(false),
+						},
+					},
+				},
+			},
+			Cpu:    pulumi.StringPtr("2048"),
+			Memory: pulumi.StringPtr("4096"),
+			Volumes: ecs.TaskDefinitionVolumeArray{
+				ecs.TaskDefinitionVolumeArgs{
+					Name: pulumi.String("cws-instrumentation-volume"),
+				},
+			},
+			ExecutionRole: &awsx.DefaultRoleWithPolicyArgs{
+				RoleArn: pulumi.StringPtr(awsEnv.ECSTaskExecutionRole()),
+			},
+			TaskRole: &awsx.DefaultRoleWithPolicyArgs{
+				RoleArn: pulumi.StringPtr(awsEnv.ECSTaskRole()),
+			},
+			Family: awsEnv.CommonNamer.DisplayName(255, pulumi.String("cws-task")),
+		}, awsEnv.WithProviders(configCommon.ProviderAWS, configCommon.ProviderAWSX))
+		if err != nil {
+			return err
+		}
+
+		_, err = ecsResources.FargateService(awsEnv, "cws-service", ecsCluster.Arn, taskDef.TaskDefinition.Arn())
+		if err != nil {
+			return err
+		}
+
+		// Export task definition's properties
+		ctx.Export(fgTaskDefArnKey, taskDef.TaskDefinition.Arn())
+
+		s.ddHostname = ddHostname
+		return nil
+	}, false)
+	s.Require().NoError(err)
+
+	s.ecsClusterArn = result.Outputs[ecsClusterArnKey].Value.(string)
+	s.ecsClusterName = result.Outputs[ecsClusterNameKey].Value.(string)
+	s.fgTaskDefArn = result.Outputs[fgTaskDefArnKey].Value.(string)
+}
+
+func (s *ECSFargateSuite) TearDownSuite() {
+	err := infra.GetStackManager().DeleteStack(s.ctx, s.stackName, nil)
+	s.Assert().NoError(err)
+}
+
+func (s *ECSFargateSuite) Test00ECSFargateReady() {
+	cfg, err := awsconfig.LoadDefaultConfig(s.ctx)
+	s.Require().NoErrorf(err, "Failed to load AWS config")
+
+	client := awsecs.NewFromConfig(cfg)
+
+	s.Run("cluster-ready", func() {
```

Comment:
The cluster and services should already be ready when the `Run` function finishes, any reason to wait again? Especially listing clusters as you already know cluster ARN?

Same for `tasks-ready`

Reply:
I didn't know that the `Run` function was already doing that, so I removed the `Test00ECSFargateReady` test entirely, thanks.